text/0000-computed-context.md

```diff
@@ -0,0 +1,329 @@
+            return self.users.filter(user => user.age < 18).length
+        },
+        numberOfPeopleOlderThan(age) {
+            return self.users.filter(user => user.age > age).length
+        }
+    }))
+
+const userStore = UserStore.create(/* */)
+
+// Every time the userStore is updated in a relevant way, log messages will be printed
+autorun(() => {
+    console.log("There are now ", userStore.numberOfChildren, " children")
+})
+autorun(() => {
+    console.log("There are now ", userStore.numberOfPeopleOlderThan(75), " pretty old people")
+})
+```
+
+</details>
+
+**MobX Computeds**
+
+https://mobx.js.org/computeds.html
+
+<details>
+<summary>
+View example
+</summary>
+
+```ts
+import { makeObservable, observable, computed, autorun } from "mobx"
+
+class OrderLine {
+    price = 0
+    amount = 1
+
+    constructor(price) {
+        makeObservable(this, {
+            price: observable,
+            amount: observable,
+            total: computed
+        })
+        this.price = price
+    }
+
+    get total() {
+        console.log("Computing...")
+        return this.price * this.amount
+    }
+}
+
+const order = new OrderLine(0)
+
+const stop = autorun(() => {
+    console.log("Total: " + order.total)
+})
+// Computing...
+// Total: 0
+
+console.log(order.total)
+// (No recomputing!)
+// 0
+
+order.amount = 5
+// Computing...
+// (No autorun)
+
+order.price = 2
+// Computing...
+// Total: 10
+
+stop()
+
+order.price = 3
+// Neither the computation nor autorun will be recomputed.
+```
+
+</details>
+
+## Detailed design
+
+### Technical Background
+
+Currently, `context` is how a state machine stores contextual data for its state. Context values are assigned via `assign(...)` actions, and the assignments can reference other context values:
+
+```js
+actions: assign({
+  fullName: (context) => `${context.firstName} ${context.lastName}`
+})
+```
+
+However, this is a manual and error-prone process, as it is easy to forget to do this whenever the relevant context values change. Assigning manually computed values is also not on-demand, and some of these computations can be expensive. 
+
+Another approach is to use a function:
+
+```js
+function fullName(context) {
+  return `${context.firstName} ${context.lastName}`;
+}
+
+const machine = createMachine({
+  // ...
+  on: {
+    someEvent: {
+      guard: (context) => fullName(context).trim().length > 100,
+      // ...
+    }
+  }
+});
+```
+
+This leads to important parts of the logic (computing the `fullName`, in this case) are now completely external to the machine, which can be less portable and is not easy to visualize in the future. Also, caching must be done manually.
+
+### Implementation
+
+The `computed: { ... }` property on the machine definition is a key-value pair of computed context keys to a function that takes in `state` and returns the computed context value from that state.
+
+Computed values _cannot_ be assigned to; they are read-only values.
```

Comment:
How about we use the term `derived state` instead of read-only? Read-only indicates they can't be assigned directly, which is good but leaves it unclear how they update. To me, the term _derived_ not only means no direct assignments but also a derivation from the assignable context values.